### ISSUE TITLE
Predicted probabilities wrong?

### ISSUE THREAD

**Issue by marsel171** [amaiya/ktrain#214]
During the prediction, it turned out that the classes with the maximum probability do not correspond to the classes after the usual prediction. Moreover, they give completely incorrect predictions of probabilities. What am I do wrong?
I use ktrain version 0.16.3
Ubuntu 18.04
Python 3.6.9

My code:
```
(x_train, y_train), (x_test, y_test), preproc = text.texts_from_df(train_df=train, 
                                                                   text_column='Text',
                                                                   label_columns=classes,
                                                                   val_pct=VAL_SPLIT, 
                                                                   max_features=MAX_WORDS, 
                                                                   maxlen=MAXLEN,
                                                                   preprocess_mode='bert',
                                                                   ngram_range=1)
model = text.text_classifier('bert', (x_train, y_train), preproc=preproc, )
learner = ktrain.get_learner(model, train_data=(x_train, y_train), val_data=(x_test, y_test))
learner.autofit(LR, EPOCH)
predictor = ktrain.get_predictor(learner.model, preproc)
predictor.save('predictor')
predictor = ktrain.load_predictor('predictor')
preds = predictor.predict(test.Text.values)
preds_proba = predictor.predict(test.Text.values,return_proba=True) 
#or preds_proba = predictor.predict_proba(test.Text.values)
```
My results:
![Screenshot from 2020-07-27 22-25-21](https://user-images.githubusercontent.com/4033005/88582939-313cf780-d058-11ea-8fb3-4da2961967d5.png)

**Comment by amaiya** [comment]
It sounds like there's something wrong with your manual calculation of the class label with the maximum probability, but you haven't shared that part of your code.  So, it's not clear what the problem is.

The `predictor.predict(test.Text.values)` (i.e., `return_proba=False`) simply computes the maximum probability and returns the corresponding class label.  If the labels returned are different than your manually-computed labels based on maximum probability, then your calculation is probably incorrect.  

Can you post more information on how your computing the values in the table  you posted?

Also, updating to the latest version of *ktrain* is recommended.

**Comment by marsel171** [comment]
This is how I calculate:
```
preds_proba['predict_proba max value']=preds_proba.max(axis=1)
preds_proba['predict_proba max value CLASS']=preds_proba.idxmax(axis=1)
```

And here is the results:
![Screenshot from 2020-07-28 10-59-32](https://user-images.githubusercontent.com/4033005/88636861-21f29400-d0c2-11ea-83e7-04951f165581.png)

First row's max is 0.28 and it corresponds to class 21. You can find it in probabilities dataframe (top).
Second row's max is 0.52 and it corresponds to class 15. 
Next few rows have maximums in class 15 too. Etc...

**Comment by amaiya** [comment]
I ran a test myself using the code you posted and everything worked correctly. Here is a self-contained demonstration:

```python
# load text data
categories = ['alt.atheism', 'soc.religion.christian','comp.graphics', 'sci.med']
from sklearn.datasets import fetch_20newsgroups
train_b = fetch_20newsgroups(subset='train', categories=categories, shuffle=True)
test_b = fetch_20newsgroups(subset='test',categories=categories, shuffle=True)
(x_train, y_train) = (train_b.data, train_b.target)
(x_test, y_test) = (test_b.data, test_b.target)

# train BERT
import ktrain
from ktrain import text
trn, val, preproc = text.texts_from_array(x_train=x_train, y_train=y_train,
                                          x_test=x_test, y_test=y_test,
                                          class_names=train_b.target_names,
                                          preprocess_mode='bert',
                                          maxlen=350)
model = text.text_classifier('bert', train_data=trn, preproc=preproc)
learner = ktrain.get_learner(model, train_data=trn, batch_size=6)
learner.fit_onecycle(2e-5, 1)
p = ktrain.get_predictor(learner.model, preproc)
preds_proba = p.predict(x_test, return_proba=True)
import pandas as pd
df = pd.DataFrame(columns=p.get_classes(), data=preds_proba)

df['predict_proba max value']=df.max(axis=1)
df['predict_proba max value CLASS']=df.idxmax(axis=1)
df['ktrain_predicted_class'] = p.predict(x_test)
df['true_class'] = [p.get_classes()[y] for y in y_test]
df.head()
```

**Comment by marsel171** [comment]
Yes, I see my bad)
Now it is right:
```
preds_proba = predictor.predict(test_new.Text.values,return_proba=True)
preds_proba = pd.DataFrame(columns = predictor.get_classes(),data=preds_proba)
```
sorry)
